### PR TITLE
Adds testing for resource server from the TCK

### PR DESCRIPTION
**Pull request by bretterer**
This should allow all the resource-server items to be tested from the TCK instead of having to download a front-end server.  This PR is mainly for internal development and should not affect any of our end users.

### CODE REVIEW

File: e2e-tests/resource-server/conf.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/* eslint import/no-unresolved:0, import/no-extraneous-dependencies:0, no-console:0 */
+const daemonUtil = require('../tools/daemon-util');
+const commonConfig = require('../tools/common-config');
+
+// Start the resource server only for implicit flow
+if (process.env.TEST_TYPE === 'implicit') {
+
+} else {
+    console.log('start');
```
Comment (vijetmahabaleshwar-okta) [inline e2e-tests/resource-server/conf.js]: Remove the console.log

File: e2e-tests/resource-server/conf.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint import/no-unresolved:0, import/no-extraneous-dependencies:0, no-console:0 */
+const daemonUtil = require('../tools/daemon-util');
+const commonConfig = require('../tools/common-config');
+
+// Start the resource server only for implicit flow
+if (process.env.TEST_TYPE === 'implicit') {
```
Comment (vijetmahabaleshwar-okta) [inline e2e-tests/resource-server/conf.js]: Don't need the if section

File: e2e-tests/tools/common-util/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                const accessToken = match && match[1];
+                if (!accessToken){
+                    return reject(new Error('Could not parse access token from URI'));
+                }
+
+                resolve(accessToken);
+            })
+        });
+    });
+};
```
Comment (vijetmahabaleshwar-okta) [inline e2e-tests/tools/common-util/index.js]: new line

File: e2e-tests/resource-server/specs/okta-resource-server-spec.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    it('returns a 401 when trying to access the /api/messages route without a token', (done) => {
+        request.get(appRoot + "/api/messages", {
+            'json': true
+        }, function (error, response, body) {
+            expect(response.statusCode).toBe(401);
+
+            done();
+        });
+    });
+});
```
Comment (vijetmahabaleshwar-okta) [inline e2e-tests/resource-server/specs/okta-resource-server-spec.js]: new line

File: e2e-tests/resource-server/conf.js (hunk 2 above)
Comment (bretterer) [reply e2e-tests/resource-server/conf.js]: Do we want to start the resource server everytime, or ignore it for implicit?

File: e2e-tests/resource-server/conf.js (hunk 1 above)
Comment (bretterer) [reply e2e-tests/resource-server/conf.js]: +1

File: e2e-tests/resource-server/conf.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint import/no-unresolved:0, import/no-extraneous-dependencies:0, no-console:0 */
+const daemonUtil = require('../tools/daemon-util');
+const commonConfig = require('../tools/common-config');
+
+module.exports.config = commonConfig.configure(Promise.all([
+    daemonUtil.startResourceServer()
```
Comment (robertjd) [inline e2e-tests/resource-server/conf.js]: Please change to 2 spaces indentation for consistency w/ rest of project

File: e2e-tests/resource-server/specs/okta-resource-server-spec.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                'bearer': this.accessToken
+            },
+            'json': true
+        }, function (error, response, body) {
+            expect(body).not.toBe(null);
+            expect(response.statusCode).toBe(200);
+            expect(Object.keys(body)).toContain('messages');
+            expect(body.messages).not.toBe(null);
+            expect(Object.keys(body.messages[0])).toContain('date');
+            expect(Object.keys(body.messages[0])).toContain('text');
```
Comment (robertjd) [inline e2e-tests/resource-server/specs/okta-resource-server-spec.js]: This string assertion feels fuzzy to me, I think we should directly assert that the response is JSON and has the keys that we expect

File: e2e-tests/resource-server/specs/okta-resource-server-spec.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+'use strict';
+const commonUtils = require('../../tools/common-util');
+const request = require('request');
+
+describe('Okta Resource Server',  () => {
+    const appRoot = `http://localhost:${browser.params.appPort}`;
+
```
Comment (vijetmahabaleshwar-okta) [inline e2e-tests/resource-server/specs/okta-resource-server-spec.js]: nit: remove empty lines

File: e2e-tests/resource-server/specs/okta-resource-server-spec.js (hunk 6 above)
Comment (robertjd) [reply e2e-tests/resource-server/specs/okta-resource-server-spec.js]: Nevermind this comment.. thought I was seeing `.contain` from string prototype

**Pull request merged**